posydon/unit_tests/utils/test_common_functions.py

```diff
@@ -0,0 +1,2273 @@
+        # examples: profile with radius
+        with warns(ClassificationWarning, match="Donor mass from the binary"+\
+                   " class object and the profile do not agree"):
+            with warns(ClassificationWarning, match="Donor radius from the"+\
+                       " binary class object and the profile do not agree"):
+                d_mass, d_radius, d_dm =\
+                 totest.get_mass_radius_dm_from_profile(star_profile[['mass',\
+                 'radius']])
+                assert np.array_equal(d_mass, star_profile['mass'])
+                assert np.array_equal(d_radius, star_profile['radius'])
+                assert np.array_equal(d_dm, star_profile['dm'])
+        # get total mass and radius
+        M = star_profile['mass'].max()
+        R = star_profile['radius'].max()
+        # examples: profile with log_R
+        d_mass, d_radius, d_dm = totest.get_mass_radius_dm_from_profile(\
+                                 star_profile[['mass', 'log_R']], m1_i = M,\
+                                 radius1 = R)
+        assert np.array_equal(d_mass, star_profile['mass'])
+        assert np.array_equal(d_radius, star_profile['radius'])
+        assert np.array_equal(d_dm, star_profile['dm'])
+        # examples: profile with radius and dm (in grams)
+        d_mass, d_radius, d_dm = totest.get_mass_radius_dm_from_profile(\
+                                 star_profile[['mass', 'dm', 'radius']],\
+                                 m1_i = M, radius1 = R)
+        assert np.array_equal(d_mass, star_profile['mass'])
+        assert np.array_equal(d_radius, star_profile['radius'])
+        # convert Msun to grams
+        d_dm = d_dm * totest.const.Msun
+        assert np.array_equal(d_dm, star_profile['dm'])
+
+    def test_get_internal_energy_from_profile(self, star_profile, monkeypatch):
+        def mock_calculate_H2recombination_energy(profile, tolerance=0.001):
+            return np.array(len(profile)*[1.0e+99])
+        def mock_calculate_recombination_energy(profile, tolerance=0.001):
+            return np.array(len(profile)*[1.0e+99])
+        # missing argument
+        with raises(TypeError, match="missing 2 required positional"+\
+                    " arguments: 'common_envelope_option_for_lambda' and"+\
+                    " 'profile'"):
+            totest.get_internal_energy_from_profile()
+        # bad input
+        with raises(AttributeError, match="'NoneType' object has no"+\
+                    " attribute 'dtype'"):
+            totest.get_internal_energy_from_profile(None, None)
+        with raises(ValueError, match="unsupported:"+\
+                    " common_envelope_option_for_lambda = test"):
+            totest.get_internal_energy_from_profile("test",\
+            star_profile[['energy']])
+        bad_profile = np.array([-1.0, -1.0, -1.0, -1.0],\
+                      dtype=([('energy', 'f8')]))
+        bad_profile2 = np.empty((2,), dtype=[(f, 'f8') for f in [\
```

Comment:
@mkruckow I realized you never actually call this bad_profile2 in a test. Is it necessary for something I'm not seeing?

Reply:
Thanks, I forgot to remove it, because the test it was needed for I instead did with the function patches.